verification/spec/connection-handshake/ch.tla

```diff
@@ -0,0 +1,188 @@
+--------------------------------- MODULE CH ---------------------------------
+
+
+VARIABLES
+    parties,            \* The state of each party.
+    inboundDatagrams    \* For each party, a buffer with any incoming datagrams. 
+
+
+Parties == {"A", "B"}
+
+ConnectionStates == { "UNINIT" }
+
+Null == "none"
+
+ConnectionIDs == {"connAtoB", "connBtoA"}
+
+
+ClientIDs == { "clientA", "clientB" }
+
+
+(***************************** ConnectionEnds *****************************
+    Ref.: https://github.com/informalsystems/ibc-rs/pull/55.
+    
+    A set of connection end records. 
+    A connection end record contains the following fields:
+    
+    - state -- a string 
+      Stores the current state of this connection end. It has one of the 
+      following values: "UNINIT", "INIT", "TRYOPEN", "OPEN".
+    
+    - connectionID -- a connection identifier
+      Stores the connection identifier of this connection end.
+    
+    - counterpartyConnectionID -- a connection identifier
+      Stores the connection identifier of the counterparty connection end.
+    
+    - clientID -- a client identifier
+      Stores the client identifier associated with this connection end. 
+      
+    - counterpartyClientID -- a client identifier
+      Stores the counterparty client identifier associated with this connection end.
+ ***************************************************************************)
+ConnectionEnd ==
+    [
+        state : ConnectionStates,
+        connectionID : ConnectionIDs \union {Null},
```

Comment:
In current specs we were normally defining special null value for each "type". In this case you might want to have NilConnectionId and NilClientId. Although I am not sure that it is needed as once you create connectionEnd object you need to set all values to not-nil one. On the other side you will probably need to have NilConnectionEnd object as initially that would be the value of the module.

Reply:
Yes, I noticed that there is a specific Null for each type. I'm sure there's a good reason to do so, but I didn't understand why (yet...) and so I didn't want to just copy/paste without making sure it is necessary & well-understood.